src/EventStore.Core/Services/Storage/ReaderIndex/AllReader.cs

```diff
@@ -57,12 +91,10 @@ public IndexReadAllResult ReadAllEventsForward(TFPos pos, int maxCount, StringFi
             long consideredEventsCount = 0;
```

Comment:
```suggestion
            long consideredEventsCount = 0L;
```
Otherwise you are converting an int to long